code/game/objects/items/trash.dm

```diff
@@ -67,4 +67,19 @@
 	burn_state = -1 //Not Burnable
 
 /obj/item/trash/attack(mob/M, mob/living/user)
-	return
\ No newline at end of file
+	return
+
+/obj/item/pornmag
+	icon = 'icons/obj/library.dmi'
+	icon_state = "pornmag"
+	name = "porn mag"
+	desc = "Look at those pixels, man!"
+	w_class = 1.0
+	var/cooldown = 0
+
+/obj/item/pornmag/attack_self(mob/user as mob)
```

Comment:
for future reference, "as mob" is not needed because you're already casting user as a mob ( mob/user)